fsspec/utils.py

```diff
@@ -388,25 +429,25 @@ def other_paths(paths, path2, is_dir=None, exists=False, flatten=False):
     return path2
 
 
-def is_exception(obj):
+def is_exception(obj: Any) -> bool:
     return isinstance(obj, BaseException)
 
 
-def isfilelike(f):
+def isfilelike(f: Any) -> TypeGuard[IO[bytes]]:
```

Comment:
Return is bool, no?

Reply:
It's a bool, but `TypeGuard` is a special annotation that says `if isfilelike(x):` then you can treat `x` like a `IO[bytes]` from then on (for flow-sensitive typing)

https://peps.python.org/pep-0647/ has the full details

Reply:
Aha, got it! So this avoids the need for something like `assert isinstance(x, io.FileIO)` elsewhere.